### DISCUSSION TITLE
Is it needed to define a custom event source name ("useEventSourceName" in @DependentResource) when having multiple dependent resources of the same type?

### DISCUSSION THREAD

**Discussion by juangon** [operator-framework/java-operator-sdk#1623] [Q&A] [answered]
Hi!
I want to use the new 4.1.1 discriminator for multiple dependent resources of the same name. As what I read in the example (https://github.com/java-operator-sdk/java-operator-sdk/blob/main/operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/multiplemanageddependentsametype/MultipleManagedDependentResourceReconciler.java) it creates a custom Event source and uses the same for all dependent resources of the same type. Is this mandatory or can we use without creating a shared event source?

Thanks!

**Comment by csviri** [comment, marked as answer]
That would be the default approach, so dependents can share the event source.

**Reply by juangon** [reply]
Ok. So no need to manually create the event source or set the event source name, right?

**Reply by csviri** [reply]
yep!

**Reply by csviri** [reply]
see sample IT for that: https://github.com/java-operator-sdk/java-operator-sdk/blob/0f9d31654886d7fd68fe26131cf9e3e6fc67be69/operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/multiplemanageddependentsametype/MultipleManagedDependentResourceReconciler.java

**Reply by juangon** [reply]
Yep I already checked that. My question was if it's mandatory to use "useEventSourcewithname" and create the custom event source by overriding "prepareEventSource" method, which you said no

**Reply by juangon** [reply]
Oh so it seems it's mandatory:

> Caused by: java.lang.IllegalArgumentException: There are multiple EventSources registered for type io.strimzi.api.kafka.model.KafkaTopic, you need to provide a name to specify which EventSource you want

**Reply by juangon** [reply]
Ok I answer myself. The failure was to add this code without "name" to the method in the discriminator:

> InformerEventSource<KafkaTopic, MyCustomResource> ies =
                (InformerEventSource<KafkaTopic, MyCustomResource>) context
                    .eventSourceRetriever().getResourceEventSourceFor(KafkaTopic.class);

After adding the name (same as "name" field in @DependentResource annotation), now it works:

> InformerEventSource<KafkaTopic, MyCustomResource> ies =
                (InformerEventSource<KafkaTopic, MyCustomResource>) context
                    .eventSourceRetriever().getResourceEventSourceFor(KafkaTopic.class, **"myCustomDependentResourceName"**);
